.github/workflows/build.yml

```diff
@@ -0,0 +1,63 @@
+name: Build and upload artifacts
+
+on:
+  push:
+    branches:
+      - main
+      - release/**
+  pull_request:
+    branches:
+      - '*'
+
+env:
+  CACHED_BUILD_PATHS: |
+    ${{ github.workspace }}/plugin-build/build/distributions/*.zip
+
+jobs:
+  job_build:
+    name: Build the plugin
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout current commit (${{ github.sha }})
+        uses: actions/checkout@v2
+
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+
+      - name: Check cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CACHED_BUILD_PATHS }}
+          key: ${{ github.sha }}
+
+      - name: Build the plugin
+        run: |
+          cd plugin-build
+          make clean
+          make compile
+
+  job_upload_artifacts:
+    name: Upload artifacts
+    needs: job_build
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout current commit (${{ github.sha }})
+        uses: actions/checkout@v2
+
+      - name: Check cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CACHED_BUILD_PATHS }}
+          key: ${{ github.sha }}
```

Comment:
I guess we dont need that either, but rather only the gradle wrapper caching

Reply:
Yup exactly. Potentially the same as:
https://github.com/getsentry/sentry-android-gradle-plugin/blob/1914df6677b4c046d711e69b28b6a1014d8612b6/.github/workflows/pre-merge.yaml#L26-L34

Reply:
By setting two jobs, the way to get the artifacts generated in the first one available in the second one is through cache. 

Why is the gradle wrapper caching required, if it's only built once (to generate the artifacts)? @cortinico the `pre-merge` workflow is run multiple times (not just once), so I don't think it's the same use case here.

Reply:
> By setting two jobs, the way to get the artifacts generated in the first one available in the second one is through cache.

If you need to share artifacts between jobs you should use https://github.com/actions/upload-artifact and https://github.com/actions/download-artifact.

Cache on CI is best-effort, it might get completely wiped between jobs and you should not rely on it for business critical files.

You can safely reuse the caching one from `pre-merge` as it will speedup the Gradle invocation, nothing more.

Reply:
Updated the build job to have a single job, to both build and upload the artifact. Good to know more about the cache on CI 😄